saleor/payment/gateway.py

```diff
@@ -54,9 +57,20 @@ def wrapped(payment: Payment, *args, **kwargs):
     return wrapped
 
 
-@payment_postprocess
+def transaction_block(fn: Callable) -> Callable:
```

Comment:
The name is not clear from my perspective but I can't find better name. Maby we should add docstring for this decorator?

Reply:
Maybe `with_locked_payment`?

Reply:
Plus a docstring.